j5/components/piezo.py

```diff
@@ -61,4 +61,14 @@ def interface_class() -> Type[PiezoInterface]:
     def buzz(self, board: Board, identifier: int,
              duration: timedelta, pitch: Pitch) -> None:
         """Queue a note to be played."""
-        self._backend.buzz(board, identifier, duration, pitch)
+        if isinstance(pitch, int):
+            frequency = pitch
+        elif isinstance(pitch, Note):
+            frequency = pitch.value
+        else:
+            raise TypeError
+
+        if frequency < 0:
+            raise ValueError
+        else:
+            self._backend.buzz(board, identifier, duration, pitch)
```

Comment:
I would prefer for the `frequency` to be passed to `self._backend.buzz` rather than `pitch` (it seems like currently we just discard `frequency` after validating it?). However since making this change affects all backends, I think it's outside the scope of this PR. I have created #122 to track this.